icu4c/source/test/intltest/tztest.cpp

```diff
@@ -2623,4 +2625,37 @@ void TimeZoneTest::TestRawOffsetAndOffsetConsistency22041() {
                      zone->getRawOffset(), raw);
     }
 }
+
+void TimeZoneTest::TestGetIanaID() {
+    const char16_t* UNKNOWN = u"Etc/Unknown";
+    static const struct {
+        const char16_t* id;
+        const char16_t* expected;
+    } TESTDATA[] = {
+        {u"",                   UNKNOWN},
+        {0,                     UNKNOWN},
+        {u"America/New_York",   u"America/New_York"},
+        {u"Asia/Calcutta",      u"Asia/Kolkata"},
+        {u"Europe/Kiev",        u"Europe/Kyiv"},
+        {u"Europe/Zaporozhye",  u"Europe/Kyiv"},
+        {u"Etc/GMT-1",          u"Etc/GMT-1"},
+        {u"Etc/GMT+20",         UNKNOWN},
+        {u"PST8PDT",            u"PST8PDT"},
+        {u"GMT-08:00",          UNKNOWN},
+        {0,                     0}
+    };
+
+    for (int32_t i = 0; TESTDATA[i].expected != 0; i++) {
+        UErrorCode sts = U_ZERO_ERROR;
+        UnicodeString inputID(TESTDATA[i].id);
+        UnicodeString ianaID;
+
+        TimeZone::getIanaID(inputID, ianaID, sts);
```

Comment:
we need to test the return value of TimeZone::getIanaID and not assume the return value is always the same as ianaID from blackbox testing point of view. Maybe assert ianaID and the return value is always contain the same?

Reply:
done